src/vulkan/engine_vulkan.cc

```diff
@@ -760,13 +760,15 @@ Result EngineVulkan::DoDrawGrid(const DrawGridCommand* command) {
   vertex_buffer->SetData(0, buf.get(), InputRate::kVertex, buf->GetFormat(), 0,
                          buf->GetFormat()->SizeInBytes());
 
-  DrawArraysCommand draw(command->GetPipeline(), *command->GetPipelineData());
+  DrawArraysCommand draw(command->GetPipeline(), *command->GetPipelineData(),
+                         command->IsTimedExecution());
```

Comment:
Similar to above, this seems like it would be better as a `SetTimedExecution` call.